paddle/fluid/framework/paddle2cinn/transform_desc.cc

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/paddle2cinn/transform_desc.h"
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "paddle/fluid/framework/paddle2cinn/vartype_utils.h"
+#include "paddle/fluid/platform/enforce.h"
+
+namespace paddle {
+namespace framework {
+namespace paddle2cinn {
+
+using PbVarType = framework::proto::VarType;
+namespace cpp = ::cinn::frontend::paddle::cpp;
+
+void TransformVarDescToCinn(framework::VarDesc *pb_desc,
+                            cpp::VarDesc *cpp_desc) {
+  cpp_desc->SetName(pb_desc->Name());
+  cpp_desc->SetType(utils::TransformVarTypeToCinn(pb_desc->GetType()));
+  cpp_desc->SetPersistable(pb_desc->Persistable());
+  if (pb_desc->Name() != "feed" && pb_desc->Name() != "fetch") {
+    cpp_desc->SetDataType(
+        utils::TransformVarDataTypeToCinn(pb_desc->GetDataType()));
+    cpp_desc->SetShape(pb_desc->GetShape());
+  }
+}
+
+void TransformVarDescFromCinn(const cpp::VarDesc &cpp_desc,
+                              framework::VarDesc *pb_desc) {
+  pb_desc->Proto()->Clear();
+  pb_desc->SetName(cpp_desc.Name());
+  pb_desc->SetType(utils::TransformVarTypeFromCinn(cpp_desc.GetType()));
+  pb_desc->SetPersistable(cpp_desc.Persistable());
+  if (cpp_desc.Name() != "feed" && cpp_desc.Name() != "fetch") {
+    pb_desc->SetShape(cpp_desc.GetShape());
+    pb_desc->SetDataType(
+        utils::TransformVarDataTypeFromCpp(cpp_desc.GetDataType()));
+  }
+}
+
+/// For OpDesc transform
+void OpInputsToCinn(framework::OpDesc *pb_desc, cpp::OpDesc *cpp_desc) {
```

Comment:
Same as above, should we make `framework::OpDesc *` as `const framework::OpDesc &` to match the style of `OpInputsFromCinn`?